base/server/python/pki/server/deployment/__init__.py

```diff
@@ -248,9 +248,20 @@ def sd_connect(self):
 
         ca_cert = os.path.join(self.mdict['pki_server_database_path'],
                                "ca.crt")
+
         if not os.path.exists(ca_cert):
-            if os.path.exists(self.mdict['pki_cert_chain_path']):
-                ca_cert = self.mdict['pki_cert_chain_path']
+
+            # if ca.crt doesn't exist, use provided cert chain
+            cert_chain_path = self.mdict['pki_cert_chain_path']
+            logger.info('Certificate chain: %s', cert_chain_path)
+
+            if cert_chain_path:
+
+                if not os.path.exists(cert_chain_path):
+                    # if cert chain is specified but doesn't exist, throw exception
+                    raise Exception('Certificate chain not found: %s' % cert_chain_path)
```

Comment:
Ahhh right. At the time, `pki_external_ca_cert_chain_path` had a default value and then we removed it during review. When I first wrote this section, it had a default value and thus we couldn't just fail if the path didn't exist (and we didn't know if we needed it to exist). +1 from me.